cardano-rosetta-server/src/server/utils/cardano/cli/cardanonode-cli.ts

```diff
@@ -13,28 +13,62 @@ export interface CardanoCli {
   submitTransaction(logger: Logger, signedTransaction: string, isMainnet: boolean): Promise<void>;
 }
 
-export const configure = (cardanoCliPath: string, networkMagic: number): CardanoCli => ({
+const SUPPORTED_ERAS = ['Tx MaryEra', 'TxSignedShelley'];
+
+/**
+ * This function returns true if DecoderErrorDeserialiseFailure is thrown.
+ *
+ * It's not the best way to detect if an era mismatch but it has been agreed
+ * it's the simplest one we can do.
+ *
+ * @param errorMessage
+ */
+const isWrongEra = (errorMessage: string): boolean => errorMessage.includes('DecoderErrorDeserialiseFailure');
```

Comment:
```suggestion
const isWrongEra = (errorMessage: string): boolean => 
  errorMessage.includes('DecoderErrorDeserialiseFailure') ||
  errorMessage.includes('The era of the node and the tx do not match');
```
When testing in an `allegra` chain, in my case was testing on `mainnet`, this error is thrown for a `Tx MaryEra` transaction type.

Reply:
I'm pushing a fix with a regex